python/ray/data/datasource/file_based_datasource.py

```diff
@@ -119,16 +129,13 @@ class FileBasedDatasource(Datasource):
     This class should not be used directly, and should instead be subclassed
     and tailored to particular file formats. Classes deriving from this class
     must implement _read_file().
-
-    If the _FILE_EXTENSION is defined, per default only files with this extension
-    will be read. If None, no default filter is used.
```

Comment:
This wasn't really true to begin with. You'd still have to specify the `partition_filter`.